src/main/java/com/ifelseco/userapi/controller/RegisterController.java

```diff
@@ -66,4 +72,53 @@ public ResponseEntity register(@RequestBody RegisterModel registerModel) throws
 
     }
 
+    @PutMapping
+    public ResponseEntity<BaseResponseModel> updateUserInfo(@RequestBody UpdateModel updateModel
+    ) {
+
+        //updateModel :)
+
+        BaseResponseModel baseResponseModel= new BaseResponseModel();
+
+
+
+        try {
+
+            User currentUser = userService.findByEmail(updateModel.getEmail());
```

Comment:
- UpdateModel should not contain email info. Because Email Update service will be  different from this User update .
- You should get email or username info from session.
- updateUserInfo method body should contain Principial object for session.

`
public ResponseEntity<BaseResponseModel> updateUser(@RequestBody UpdateModel updateModel, Principal principal) {

User currentUser = userService.findByUsername(principal.getName());

`